sandbox_dns_unix.go

```diff
@@ -313,8 +315,31 @@ func (sb *sandbox) rebuildDNS() error {
 	// external v6 DNS servers has to be listed in resolv.conf
 	dnsList = append(dnsList, resolvconf.GetNameservers(currRC.Content, types.IPv6)...)
 
-	// Resolver returns the options in the format resolv.conf expects
-	dnsOptionsList = append(dnsOptionsList, sb.resolver.ResolverOptions()...)
+	// If the user entered a ndots value > 0 remember that. container's resolv.conf
+	// will only have ndots:0 for embedded DNS server to get unqualified names.
+	resOptions := sb.resolver.ResolverOptions()
+	for i, option := range dnsOptionsList {
+		if strings.Contains(option, "ndots") {
+			parts := strings.Split(option, ":")
+			if len(parts) == 0 || len(parts) > 2 {
+				return fmt.Errorf("invalid ndots option %v", option)
+			}
+			if num, err := strconv.Atoi(parts[1]); err != nil {
+				return fmt.Errorf("invalid ndots option %v", option)
+			} else if num > 0 {
+				sb.ndotsSet = true
+			}
+			for _, resOpt := range resOptions {
+				if strings.Contains(resOpt, "ndots") {
```

Comment:
What I mean is that, if `ResolverOptions()` in the future removes the ndots, then the current logic will change and the outer for loop will run through all the `dnsOptionsList` even if the `ndots` check has passed once. While in the current state since `ResolverOptions` has a static definition of ndots, the outer loop deterministically exits.